File: 6kyu/mini_string_fuck.hs
module MiniStringFuck where

import Data.Char

myFirstInterpreter :: String -> String
myFirstInterpreter code = f code 0
  where
    f "" _ = ""
    f ('+':xs) n = f xs ((n + 1) `mod` 256)
    f ('.':xs) n = chr n : f xs n
    f (_:xs) n = f xs n

File: 6kyu/mini_string_fuck.js
function myFirstInterpreter(code) {
  let n = 0;
  let ans = '';
  for (const cmd of code) {
    if (cmd === '+') {
      n += 1;
      n %= 256;
    } else if (cmd === '.') {
      ans += String.fromCharCode(n);
    }
  }
  return ans;
}
